### PR TITLE
added plotly 4.14.3 and dependencies

### PR DESCRIPTION
**Pull request by dmondev**
Adds the latest version of plotly as a pyodide package

Closes https://github.com/pyodide/pyodide/issues/1267

### CODE REVIEW

**Comment by dmondev** [comment]
@rth , do you think this can be merged ?

**Comment by rth** [comment]
Actually `retrying` failed to import both in test-packages-firefox and test-packages-chrome. Could you please merge master in, to see if it will still be an issue? Thanks!

**Comment by dmondev** [comment]
Sure, @rth, will do !

**Comment by hoodmane** [comment]
If people wish to use plotly v5.0 once it comes out, how do they do that? Will `micropip.install("plotly")` install this version? Can they say something like `micropip.install("plotly>=5.0")`?

**Comment by rth** [comment]
> Can they say something like micropip.install("plotly>=5.0")?

That should have worked. But currently there is a known bug that makes pyodide included packages always installed if available irrespective of the version https://github.com/pyodide/pyodide/issues/725

**Comment by hoodmane** [comment]
Okay I opened PR #1457 for that.

**Comment by dmondev** [comment]
I'm getting the same error when trying to build from master without this PR:

```
Traceback (most recent call last):
  File "/root/repo/emsdk/emsdk/upstream/emscripten/tools/system_libs.py", line 1707, in retrieve
    import requests
ModuleNotFoundError: No module named 'requests'
```

**Comment by rth** [comment]
Yes, it's general issue with downloads from bintray. That error is misleading. 

I'll make a PR to backport https://github.com/emscripten-core/emscripten/pull/13869 which should fix it.

**Comment by hoodmane** [comment]
@dmondev You should merge master here, rth merged the emscripten patch.

**Comment by dmondev** [comment]
> @dmondev You should merge master here, rth merged the emscripten patch.

Thanks, @hoodmane, just rebased and pushed.

**Comment by hoodmane** [comment]
Now it imported successfully on Chrome but in Firefox it failed with `ModuleNotFoundError: No module named 'six'`. I reran CI again.

**Comment by rth** [comment]
The `six` import failure is genuine, `retrying` does have a dependency on `six` https://github.com/conda-forge/retrying-feedstock/blob/98d1adf530bd3aff64a6bac53b59cb2ce1b4ed2e/recipe/meta.yaml#L30 so it would need to be added

**Comment by hoodmane** [comment]
But `import six` works in firefox in https://pyodide-cdn2.iodide.io/dev/full/console.html

File: packages/retrying/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package:
+  name: retrying
+  version: 1.3.3
+source:
+  sha256: 08c039560a6da2fe4f2c426d0766e284d3b736e355f8dd24b37367b0bb41973b
+  url: https://files.pythonhosted.org/packages/44/ef/beae4b4ef80902f22e3af073397f079c96969c69b2c7d52a57ea9ae61c9d/retrying-1.3.3.tar.gz
+test:
```
Comment (hoodmane) [inline packages/retrying/meta.yaml:7]:
```suggestion
requirements:
  run:
    - six
test:
```

**Comment by dmondev** [comment]
Sorry missed this, thanks, @hoodmane !

**Pull request merged**